### PR TITLE
Query transaction table

### PR DESCRIPTION
**Pull request by samau3**
1. Added a frontend to allow user to make buy transactions that will query and update the database
2. Added a success page upon successful buy transaction, will need to add error page upon failed transaction
3. Added placeholder sell and quote functions

### CODE REVIEW

File: app/src/App.tsx [hunk 1]
```diff
@@ -90,6 +92,8 @@ function App() {
             path='profile'
             element={<Profile handleLogout={handleLogout} />}
           />
+          <Route path='trading' element={<Trading />} />
+          <Route path='success' element={<Success />} />
```
Comment (li-andy) [inline app/src/App.tsx:96]:
```suggestion
          <Route path='trading' element={<TradingPage />} />
          <Route path='success' element={<SuccessPage />} />
```

File: app/src/helpers/PaperTraderApi.ts [hunk 2]
```diff
@@ -33,6 +33,26 @@ class PaperTraderApi {
       .userPortfolio;
     return result;
   }
+
+  static async buyStock(ticker: string, quantity: number) {
+    const config = {
+      headers: {
+        Authorization: `Bearer ${PaperTraderApi.token}`,
+      },
+    };
```
Comment (li-andy) [inline app/src/helpers/PaperTraderApi.ts:42]: seems like this is shared among other routes—consider pulling this out

File: server/src/routes/stock.ts [hunk 3]
```diff
@@ -51,6 +53,8 @@ router.post('/buy', async (req, res, next) => {
     const price: number = quote.c;
     const total = Number((price * qty).toFixed(2));
 
+    await Transaction.buyTransction(ticker, qty, price, res.locals.user);
```
[the remark this answers is not in the document]
Comment (li-andy) [reply server/src/routes/stock.ts:56]: ` Transaction.buy()` feels adequate (`Transaction.buyTransaction` feels a bit redundant)

File: server/src/routes/stock.ts [hunk 4]
```diff
@@ -76,9 +80,10 @@ router.post('/sell', async (req, res, next) => {
       throw new BadRequestError('Invalid Stock Ticker');
     }
 
-    const price: number = quote.c;
+    const price: number = -quote.c;
```
Comment (li-andy) [inline server/src/routes/stock.ts:83]: what if the user has insignificant funds?

File: server/src/routes/stock.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (samau3) [reply server/src/routes/stock.ts:56]: Renamed the `buy` method so it's not as redundant

File: app/src/App.tsx (hunk 1 above)
Comment (samau3) [reply app/src/App.tsx:96]: Renamed the functions so they're more indicative of what they represent

**Review by li-andy — APPROVED** [review]

**Pull request merged**